c10/cuda/CUDAAllocatorConfig.h

```diff
@@ -1,164 +1,33 @@
 #pragma once
 
+#include <c10/core/AllocatorConfig.h>
 #include <c10/cuda/CUDAMacros.h>
 #include <c10/util/Exception.h>
 #include <c10/util/env.h>
 
-#include <atomic>
-#include <cstddef>
-#include <cstdlib>
-#include <mutex>
-#include <string>
-#include <vector>
-
 namespace c10::cuda::CUDACachingAllocator {
 
-enum class Expandable_Segments_Handle_Type : int {
-  UNSPECIFIED = 0,
-  POSIX_FD = 1,
-  FABRIC_HANDLE = 2,
-};
-
-// Environment config parser
-class C10_CUDA_API CUDAAllocatorConfig {
+// Keep this for backwards compatibility
+class C10_CUDA_API CUDAAllocatorConfig
+    : public CachingAllocator::AllocatorConfig {
  public:
-  static size_t max_split_size() {
-    return instance().m_max_split_size;
-  }
-  static double garbage_collection_threshold() {
-    return instance().m_garbage_collection_threshold;
-  }
-
   static bool expandable_segments() {
-#ifndef PYTORCH_C10_DRIVER_API_SUPPORTED
-    if (instance().m_expandable_segments) {
-      TORCH_WARN_ONCE("expandable_segments not supported on this platform")
-    }
-    return false;
-#else
-    return instance().m_expandable_segments;
-#endif
-  }
-
-  static Expandable_Segments_Handle_Type expandable_segments_handle_type() {
```

Comment:
This method is landed recently. And nobody uses it inthe  community. refer to https://github.com/search?q=set_expandable_segments_handle_type&type=code